packages/rspeedy/core/src/cli/build.ts

```diff
@@ -23,15 +24,24 @@ export async function build(
   // We always exit on CI since `sass-embedded` will have child_processes that never exit.
   // Otherwise, we do not exit when Rsdoctor is enabled.
   const shouldExit = process.env['RSDOCTOR'] !== 'true' || isCI()
+  const isWatch = buildOptions.watch ?? false
 
   try {
     const { createRspeedyOptions } = await init(cwd, buildOptions)
 
     const rspeedy = await createRspeedy(createRspeedyOptions)
 
-    // TODO: support `rspeedy build --watch`
-    const { close } = await rspeedy.build()
-    await close()
+    const { close } = await rspeedy.build({
+      watch: isWatch,
+    })
+
+    if (isWatch) {
+      process.on('SIGINT', () => {
```

Comment:
It seems like we already have the `SIGINT` handler at [`exit.ts`](https://github.com/lynx-family/lynx-stack/blob/52db511fe59430667228dee43c635839ad2a5c18/packages/rspeedy/core/src/cli/exit.ts#L33).

https://github.com/lynx-family/lynx-stack/blob/52db511fe59430667228dee43c635839ad2a5c18/packages/rspeedy/core/src/cli/exit.ts#L33

We don't need another handler here. Instead, we need to setup listeners for configuration changes. Like `rspeedy dev` or `rsbuild build --watch` does:

https://github.com/web-infra-dev/rsbuild/blob/6274fdae203f6624cdce44cedfc66f525dd8d9f2/packages/core/src/cli/commands.ts#L135-L137